src/quickstart.ts

```diff
@@ -0,0 +1,111 @@
+import * as fsNode from 'fs';
+import * as path from 'path';
+
+import {fs} from './pr';
+
+export enum ProjectType {
+  Library,
+  Exectable
+}
+
+export interface ProjectTypeDesciptor {
+  label: string;
+  type: ProjectType;
+  description: string;
+}
+
+export const projectTypeDescriptions: ProjectTypeDesciptor[] = [
+  {label: 'Library', type: ProjectType.Library, description: 'Create a library'},
+  {label: 'Executable', type: ProjectType.Exectable, description: 'Create an executable'}
+];
+
+export class CMakeQuickStart {
```

Comment:
It would be simpler to have a free function that implements `createProject()` and also takes the `workingPath` as a parameter, since the only use of this class is to be created and immediately call `createProject()`.

Reply:
Try this design:

Have an `interface QuickStartParams` that includes things like the `ProjectType`, the `name` string, the source directory, etc. Have an `async` function that takes the params and performs the whole quickstart. You can then test the quickstart without opening any project directory or mocking any UI elements.

As a similar change, you can create an `async getUserQuickStartParams()` function that performs all the UI requests to generate a `QuickStartParams` object. If you make this function simple enough, no testing of it will be necessary.

@ytimenkov, opinions?

Reply:
yes, I will move the path generation into the create functions for cmake and executeable or library. This allows to remove the class and add them as namespace functions.

Reply:
I don't understand the need of QuickStartParams. I think we could move  quickstart UI parts out of the CMakeTools class to arrive the single responsibility. At the moment it is still part because we need to call the configure command.  do we have a interface for all commands? Then we code use dependency injection. 

The extension tests I have created are only used to test the UI integration. The unit tests need a workspace to simplify clean up.
I do not see a way to reduce the tests. The strong types have already reduce the count of tests.

Reply:
- [ ] Extract Quickstart UI workflow from CMakeTools-Class into separate function 
This function is using the interface CMakeToolsApi and class CMakeQuickstart to access needed backend behavior. 
All UI interactions could be moved into callbacks. This allows to test the workflow without UI interaction. The workflow class will not replace the UI tests, it will reduce only the count of tests and the order of user interaction:

1. CMakeList present
2. project name
3. project type
4. Open source file
5. configure project